src/test/resources/optimized-query-for-filter.adoc

```diff
@@ -0,0 +1,897 @@
+WITH movie, movie_reviewedBy_some
+
+MATCH (movie_reviewedBy_some)<-[:`FOLLOWS`]-(movie_reviewedBy_some_followedBy_some:`Person`)
+WITH movie, movie_reviewedBy_some_followedBy_some
+
+MATCH
+  (movie_reviewedBy_some_followedBy_some)-[:`REVIEWED`]->(movie_reviewedBy_some_followedBy_some_reviewedMovies_some:`Movie`)
+WHERE movie_reviewedBy_some_followedBy_some_reviewedMovies_some.released >= $movie_reviewedBy_some_followedBy_some_reviewedMovies_some_released_gte
+WITH DISTINCT movie
+
+RETURN DISTINCT movie { .title } AS movie
+----
+
+.Movie Example result
+|===
+|title
+
+|Unforgiven
+
+|===
+
+=== Filter single (n..m)
+
+.Query configuration
+[source,json,config=true]
+----
+{  "optimizedQuery": true }
+----
+
+.GraphQL-Query
+[source,graphql]
+----
+{
+  movie(filter: {
+    directedBy_single:{
+      name: "Rob Reiner"
+    }
+  }) {
+    title
+  }
+}
+----
+
+.Cypher params
+[source,json]
+----
+{
+  "movie_directedBy_single_name":"Rob Reiner"
+}
+----
+
+.Cypher
+[source,cypher]
+----
+MATCH (movie:`Movie`)
+WITH movie
+MATCH (movie)<-[:`DIRECTED`]-(movie_directedBy_single:`Person`)
+WHERE movie_directedBy_single.name = $movie_directedBy_single_name
+WITH movie,
+	size((movie)<-[:`DIRECTED`]-(:`Person`)) AS movie_directedBy_single_total,
+	count(DISTINCT movie_directedBy_single) AS movie_directedBy_single_count
+WHERE (movie_directedBy_single_total = movie_directedBy_single_count
+	AND movie_directedBy_single_total = 1)
+
+WITH DISTINCT movie
+
+RETURN DISTINCT movie { .title } AS movie
+----
+
+.Movie Example result
+|===
+|title
+
+|A Few Good Men
+|Stand By Me
+|When Harry Met Sally
+|===
+
+=== Filter single (n..1)
+
+.Query configuration
+[source,json,config=true]
+----
+{  "optimizedQuery": true }
+----
+
+.GraphQL-Query
+[source,graphql]
+----
+{
+  movie(filter: {
+    directedBySingle:{
+      name: "Rob Reiner"
+    }
+  }) {
+    title
+  }
+}
+----
+
+.Cypher params
+[source,json]
+----
+{
+  "movie_directedBySingle_name":"Rob Reiner"
+}
+----
+
+.Cypher
+[source,cypher]
+----
+MATCH (movie:`Movie`)
+WITH movie
+MATCH (movie)<-[:`DIRECTED`]-(movie_directedBySingle:`Person`)
+WHERE movie_directedBySingle.name = $movie_directedBySingle_name
+WITH DISTINCT movie
+
+RETURN DISTINCT movie { .title } AS movie
+----
+
+.Movie Example result
+|===
+|title
+
+|A Few Good Men
+|Stand By Me
+|When Harry Met Sally
+|===
+
+=== Filter none (n..m)
+
+.Query configuration
+[source,json,config=true]
+----
+{  "optimizedQuery": true }
+----
+
+.GraphQL-Query
+[source,graphql]
+----
+{
+  movie(filter: {
+    directedBy_none:{
+      name: "Lilly Wachowski"
+    }
+  }) {
+    title
+  }
+}
+----
+
+.Cypher params
+[source,json]
+----
+{
+  "movie_directedBy_none_name":"Lilly Wachowski"
+}
+----
+
+.Cypher
+[source,cypher]
+----
+MATCH (movie:`Movie`)
+WITH movie
+WHERE size([(movie)<-[:`DIRECTED`]-(movie_directedBy_none:`Person`) WHERE movie_directedBy_none.name = $movie_directedBy_none_name | true]) = 0
```

Comment:
is this the old implementation? the new one would probably use an optional match and a count of 0 ?

Reply:
:+1: